src/main/java/uk/nhs/ctp/service/CdssSupplierService.java

```diff
@@ -61,6 +64,12 @@ public CdssSupplier getCdssSupplier(Long id) {
         .orElseThrow(EMSException::notFound);
   }
 
+  public Optional<CdssSupplier> findCdssSupplierByBaseUrl(String baseUrl) {
+    return cdssSupplierRepository.findAllBySupplierId(authService.requireSupplierId()).stream()
```

Comment:
As this is in the path of every request, I'd prefer to add an index on (supplierId, baseURL) and a query for looking up by those fields.

Reply:
Did consider that but didn't think there'd be many registered suppliers returned anyway. I'll add it